core/PRU.cpp

```diff
@@ -384,12 +394,9 @@ int PRU::initialise(BelaHw newBelaHw, int pru_num, bool uniformSampleRate, int m
 	pru_number = pru_num;
 
 	/* Allocate and initialize memory */
-	prussdrv_init();
-	if(prussdrv_open(PRU_EVTOUT_0)) {
-		fprintf(stderr, "Failed to open PRU driver\n");
-		return 1;
-	}
-	pruMemory = new PruMemory(pru_number, context);
+
+	prumanager->start();
```

Comment:
`prussdrv_open()` should be replaced by  `new PruMemory()`  and  `prussdrv_exec_code` should be replaced by `pruManager->start()`.

Reply:
>prussdrv_open() should be replaced by new PruMemory

Are you sure? I think what `prussdrv_open()` partially does is checks if it "failed to open PRU driver" and accordingly returns 1 or 0 and then accordingly `if-else` works. However we would be eliminating the return _inside_ `if-else` it if we were to be placed it in the constructor, is that okay?

Reply:
sorry, it should have been `new PruManager`